astropy/io/misc/hdf5.py

```diff
@@ -214,6 +213,22 @@ def _encode_mixins(tbl):
     return encode_tbl
 
 
+def check_column_encoding(table):
+    """
+    Table with native py3 strings can't be stored in a hdf5 format so if
+    table has py3 strings then a copy of table is made and stored in temp.
+    the copy stored in temp has columns in bytes unicode.
+    """
+    colnames = [name for name in table.colnames if table.columns[name].info.dtype.kind == 'U']
+    if not colnames:
+        return table
+    else:
+        temp = table.copy(copy_data=False)
+        for column_name in colnames:
+            temp.replace_column(column_name, np.char.encode(table.columns[column_name], "utf-8"))
```

Comment:
@taldcroft - I wonder whether we should maybe update the ``convert_unicode_to_bytestring`` Table method so that it can take an encoding (that defaults to ASCII), and if it's not ASCII, it can do something like the code above?